docs/extend/modular-resources/create.md

```diff
@@ -0,0 +1,511 @@
+
+    # move_straight: unimplemented
+    async def move_straight(self, distance: int, velocity: float, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+        pass
+
+    # spin: unimplemented
+    async def spin(self, angle: float, velocity: float, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+        pass
+
+    # set_velocity: unimplemented
+    async def set_velocity( self, linear: Vector3, angular: Vector3, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+        pass
+
+    # set_power: set the linear and angular velocity of the left and right motors on the base
+    async def set_power(self, linear: Vector3, angular: Vector3, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+
+        # stop the base if absolute value of linear and angular velocity is less than .01
+        if abs(linear.y) < 0.01 and abs(angular.z) < 0.01:
+            return self.stop(extra=extra, timeout=timeout)
+
+        # use linear and angular velocity to calculate percentage of max power to pass to SetPower for left & right motors
+        sum = abs(linear.y) + abs(angular.z)
+
+        self.left.set_power(power=((linear.y - angular.z) / sum), extra=extra, timeout=timeout)
+        self.right.set_power(power=((linear.y + angular.z) / sum), extra=extra, timeout=timeout)
+
+    # stop: stop the base from moving by stopping both motors
+    async def stop(self, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):
+        self.left.stop(extra=extra, timeout=timeout)
+        self.right.stop(extra=extra, timeout=timeout)
+
+    # is_moving: check if either motor on the base is moving with motors' is_powered
+    async def is_moving(self, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs) -> bool:
+        return self.left.is_powered(extra=extra, timeout=timeout)[0] or self.right.is_powered(extra=extra, timeout=timeout)[0]
+```
+
+{{% /expand%}}
+
+<file>_\_init__.py</file>
+
+{{%expand "Click to view example code from a Python module registering the mybase custom model and API" %}}
+
+``` python {class="line-numbers linkable-line-numbers"}
+"""
+This file registers the MyBase model with the Python SDK.
+"""
+
+from viam.components.motor import *
+from viam.components.base import Base
+from viam.resource.registry import Registry, ResourceCreatorRegistration
+
+from .my_base import MyBase
+
+Registry.register_resource_creator(Base.SUBTYPE, MyBase.MODEL, ResourceCreatorRegistration(MyBase.new_base, MyBase.validate_config))
+```
+
+{{% /expand%}}
+
+### Code a main program that serves as the entry point file to the module
+
+[Go Client SDK](https://pkg.go.dev/go.viam.com/rdk):
+
+<file>main.go</file>
+
+{{%expand "Click to view example code from a Go module's entry point file initializing the mybase custom model and API from the registry" %}}
```

Comment:
```suggestion
{{%expand "Click to view example code from a Go module's entry point file initializing the mybase custom model from the registry" %}}
```
As above--are they actually registering an API or just a model that uses that API?

Reply:
registering helpers w SDK for both I think, it's confusing

Reply:
Hey for the sake of clarity etc. instead of editing my suggestions themselves, mind making a new suggestion on the thread and copy pasting from mine if you want?

What is an API helper? I'm still thinking "API" should come out of here

Reply:
But I do like taking "from the registry" off too since it's long